database/gormdb_test.go

```diff
@@ -751,6 +751,54 @@ func TestGormDBUpdateCourse(t *testing.T) {
 	}
 }
 
+func TestGormDBCourseUniqueContraint(t *testing.T) {
+	// Test that a course with the same code and year cannot be created.
+	db, cleanup := qtest.TestDB(t)
+	defer cleanup()
+	wantCourse := &pb.Course{
+		Name:           "Test Course",
+		Code:           "DAT100",
+		Year:           2017,
+		Tag:            "Spring",
+		Provider:       "github",
+		OrganizationID: 1235,
+	}
+	course := &pb.Course{
+		Name:           "Test Course 2",
+		Code:           "DAT100",
+		Year:           2017,
+		Tag:            "Spring",
+		Provider:       "github",
+		OrganizationID: 1234,
+	}
+
+	remoteID := &pb.RemoteIdentity{Provider: wantCourse.Provider, RemoteID: 10, AccessToken: "token"}
+	admin := qtest.CreateUserFromRemoteIdentity(t, db, remoteID)
+
+	if err := db.CreateCourse(admin.ID, wantCourse); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := db.CreateCourse(admin.ID, course); err == nil {
```

Comment:
Suggest to move the comment outside to clarify what is expected to happen. (General English language tip: Whenever you see yourself writing _this_, stop and think if you can be more specific.)
And check that we get the expected error type.
```go
// CreateCourse should fail because the unique constraint (course.code, course.year) is violated
if err := db.CreateCourse(admin.ID, course); err != nil && !errors.Is(err, database.ErrCourseExists) {
	t.Fatal(err)
}
```